tika-parsers/tika-parsers-classic/tika-parsers-classic-modules/tika-parser-zip-commons/src/main/java/org/apache/tika/detect/zip/DefaultZipContainerDetector.java

```diff
@@ -241,12 +251,27 @@ private MediaType detect(ZipArchiveEntry zae, ZipArchiveInputStream zis,
     }
 
     private MediaType finalDetect(StreamingDetectContext detectContext) {
-        for (ZipContainerDetector d : zipDetectors) {
+        for (ZipContainerDetector d : getDetectors()) {
             MediaType mt = d.streamingDetectFinal(detectContext);
             if (mt != null) {
                 return mt;
             }
         }
         return MediaType.APPLICATION_ZIP;
     }
+
+    private List<ZipContainerDetector> getDetectors() {
+        List<ZipContainerDetector> zipDetectors = new ArrayList<>();
+        if (loader != null) {
+            zipDetectors.addAll(loader.loadDynamicServiceProviders(ZipContainerDetector.class));
+            if (staticZipDetectors == null) {
```

Comment:
Why does this class handle the 'static' services differently than the other classes?

Reply:
because the other classes have a superclass that returns the static detectors in getDetectors() but this class does not, so it needs to store them in staticZipDetectors

Reply:
Do you have other examples where the static service providers are loaded lazily? I thought that the static providers get loaded eagerly everywhere else.

Reply:
Do you have other examples where the static service providers are loaded lazily? I thought that the static providers get loaded eagerly everywhere else.

Reply:
No I don't - I will move it to the constructor if it problematic to load lazily